### PR TITLE
Stylistic changes from Spark maintainers

### PR DESCRIPTION
**Pull request by ajacques**
Improvements as requested by Spark maintainers in apache/spark#21889

### CODE REVIEW

**Comment by mallman** [comment]
Thanks for your PR, @ajacques. I will review in earnest.

**Comment by mallman** [comment]
@ajacques Thanks for addressing the merge conflicts. I've been occupied with some other work-related responsibilities today. I'll try to get to this tomorrow (Friday).

Cheers.

**Comment by mallman** [comment]
@ajacques Should I give co-attribution to @HyukjinKwon for this work? As-is it only has your name on it.

**Comment by HyukjinKwon** [comment]
I'm okay in any way.

**Comment by mallman** [comment]
@ajacques I removed my changes to `ParquetTest.scala` upstream. Can you do a rebase and verify that your PR includes no changes to `ParquetTest.scala`? Thanks.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/SelectedFieldSuite.scala [hunk 1]
```diff
@@ -27,309 +27,373 @@ import org.apache.spark.sql.catalyst.parser.CatalystSqlParser
...
+    val testRelation = LocalRelation(nestedComplex.toAttributes)
+    assertResult(None)(unapplySelect("col1", testRelation))
+    assertResult(None)(unapplySelect("col1 as foo", testRelation))
+    assertResult(None)(unapplySelect("col2", testRelation))
   }
 
-  info("For a relation with schema\n" + indent(schema.treeString))
+  //  |-- col1: string (nullable = false)
+  //  |-- col2: struct (nullable = true)
+  //  |    |-- field1: integer (nullable = true)
```
Comment (mallman) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/SelectedFieldSuite.scala]: I don't see `field1` in `structOfArray`. Please remove it here or add it there.

**Review by mallman — CHANGES_REQUESTED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/execution/SelectedFieldSuite.scala (hunk 1 above)
Comment (ajacques) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/SelectedFieldSuite.scala]: You're correct, good catch. I've removed this one. `field1` is belongs in the previous schema.

**Comment by ajacques** [comment]
> @ajacques Should I give co-attribution to @HyukjinKwon for this work? As-is it only has your name on it.

Sure, let's do that. HyukjinKwon's comments were a valuable resource in this pull request. Is there anything I need to do on my end for this?

> @ajacques I removed my changes to ParquetTest.scala upstream. Can you do a rebase and verify that your PR includes no changes to ParquetTest.scala? Thanks.

Rebased, and checked that there's no changes to that file now.

**Comment by mallman** [comment]
Merged. Thank you.

**Pull request merged**